docs/tutorial2.md

```diff
@@ -135,24 +134,24 @@ virtual void OnUnitIdle(const Unit& unit) final {
 Now we just need to implement FindNearestMineralPatch and we can fix our lazy SCV.
 
 ```C++
-bool FindNearestMineralPatch(const Point2D& start, uint64_t& target) {
+const Unit* FindNearestMineralPatch(const Point2D& start) {
     Units units = Observation()->GetUnits(Unit::Alliance::Neutral);
     float distance = std::numeric_limits<float>::max();
+    const Unit* target = nullptr;
     for (const auto& u : units) {
-        if (u.unit_type == UNIT_TYPEID::NEUTRAL_MINERALFIELD) {
-            float d = DistanceSquared2D(u.pos, start);
+        if (u->unit_type == UNIT_TYPEID::NEUTRAL_MINERALFIELD) {
+            float d = DistanceSquared2D(u->pos, start);
             if (d < distance) {
                 distance = d;
-                target = u.tag;
+                target = u;
             }
         }
     }
-
+    //If we never found one return false;
     if (distance == std::numeric_limits<float>::max()) {
-        return false;
+        return target;
```

Comment:
If distance never changes that means target will be nullptr. We don't need this check anymore. Just return target at the end, it will be nullptr in the case that distance is max.